src/Microsoft.DotNet.Wpf/src/System.Windows.Primitives/Windows/Win32/Foundation/ComScope.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Windows.Win32.System.Com;
+
+namespace Windows.Win32.Foundation;
+
+/// <summary>
+///  Lifetime management struct for a native COM pointer. Meant to be utilized in a <see langword="using"/> statement
+///  to ensure <see cref="IUnknown.Release"/> is called when going out of scope with the using.
+/// </summary>
+/// <remarks>
+///  <para>
+///   This struct has implicit conversions to T** and void** so it can be passed directly to out methods.
+///   For example:
+///  </para>
+///  <code>
+///   using ComScope&lt;IUnknown&gt; unknown = new(null);
+///   comObject-&gt;QueryInterface(&amp;iid, unknown);
+///  </code>
+///  <para>
+///   Take care to NOT make copies of the struct to avoid accidental over-release.
+///  </para>
```

Comment:
💡 Roslyn's `[NonCopyable]` can be used for this

Reply:
Can you provide more information on where this attribute is defined and which analyzer package needs to be installed to leverage it?

Reply:
I'll look into it, thanks.